src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -97,17 +98,16 @@ class SubgraphStaticExecutor : public Subgraph::SubgraphExecutor {
 protected:
     typedef void (*kernel)(const void*, const void*);
 
-    inline void init_call_args(jit_snippets_call_args& call_args, const std::vector<MemoryPtr>& srcMemPtrs, const std::vector<MemoryPtr>& dstMemPtrs) {
+    inline void init_call_args(jit_snippets_call_args& call_args, const std::vector<MemoryPtr>& srcMemPtrs,
+                               const std::vector<MemoryPtr>& dstMemPtrs, size_t ithr) {
         for (size_t i = 0; i < srcMemPtrs.size(); i++)
             call_args.src_ptrs[i] = srcMemPtrs[i]->getDataAs<const uint8_t>() + m_start_offset_in[i];
 
         for (size_t i = 0; i < dstMemPtrs.size(); i++)
             call_args.dst_ptrs[i] = dstMemPtrs[i]->getDataAs<uint8_t>() + m_start_offset_out[i];
 
-        if (m_buffer_scratchpad_size > 0) {
-            call_args.buffer_scratchpad_ptr =
-                    reinterpret_cast<uint8_t*>(m_buffer_scratchpad.data()) + parallel_get_thread_num() * m_buffer_scratchpad_size;
-        }
+        if (m_buffer_scratchpad_size > 0)
+            call_args.buffer_scratchpad_ptr = m_buffer_scratchpad->getDataAs<uint8_t>() + ithr * m_buffer_scratchpad_size;
```

Comment:
Do you think it's worth to create a corresponding inline function in the base class to avoid duplication?
Like `inline void update_scratchpad_ptr(jit_snippets_call_args& call_args, size_t ithr)` or even better:
`inline void update_scratchpad_ptr(void*& scratchpad_ptr, size_t ithr)`

Reply:
Yeah, good idea. Done, thanks